ui/ui_v1/scripts/process_query.js

```diff
@@ -0,0 +1,236 @@
+//json object that contains all aspects and table's data
+var JsonObj = {
+  'intent': 'null',
+  'table': 'null',
+  'rowRange': 'null',
+  'metric': 'null',
+  'dimensions': 'null',
+  'summaryOperator': 'null',
+  'isAsc': 'null',
+  'k': 'null',
+  'slices': 'null',
+  'dateRange': 'null',
+  'timeGranularity': 'null'
+}
+
+//function to set keys of JsonObj, called from client side javascript
+function createJsonObject(intent,tableData,metric,summaryOperator,isAsc,k,
+  dimensions,slices,dateRange,timeGranularity,outputSheet) {
+
+  /*
+   * set intent 
+   * JsonObj.intent is a string storing intent name
+   */
+  JsonObj.intent = intent;
+  
+  /*
+   * set table and rowRange
+   * JsonObj.table is a 2d array consisting of sheet's data
+   */
+  JsonObj.table = getTable(tableData);
+  
+  /*
+   * set metric
+   * JsonObj.metric is a string storing column name
+   */
+  JsonObj.metric = metric;
+  
+  /*
+   * set dimension
+   * JsonObj.dimensions is a 1d array of strings storing name of output columns
+   */
+  if(dimensions.length > 0)
+    JsonObj.dimensions = dimensions;
+  
+  /*
+   * set summary
+   * JsonObj.summaryOperator is a string storing the name of operation to be applied
+   */
+  JsonObj.summaryOperator = summaryOperator;
+  
+  /* 
+   * set sortOrder
+   * JsonObj.isAsc is a boolean variable which stores true for ascending order
+   */
+  JsonObj.isAsc = isAsc;
+  
+  /* 
+   * set limit
+   * JsonObj.k is an Integer storing K 
+   * (if user doesn't specify values of k -5000000 default)
+   */
+  JsonObj.k = k;
+   
+  /*
+   * set slice
```

Comment:
Explain what is a slice here, or link to a doc or something that has an explanation.